### ISSUE TITLE
Possible crash, freezing and bugs on Swamp bloodsucker cutscene

### ISSUE THREAD

**Issue by SurDno** [Decane/SRP#20]
If all squad members were killed after bloodsucker cutscene started, but before bloodsucker finally spawned, the game may crash with this log:

```
Expression    : fatal error
Function      : CScriptEngine::lua_error
File          : E:\priquel\sources\engine\xrServerEntities\script_engine.cpp
Line          : 180
Description   : <no expression>
Arguments     : LUA error: No such operator defined

lua stack:
  [ 1|-1] (no value):  
```

Here's the full log if that's not enough: [xray_Харлам.log](https://github.com/Decane/SRP/files/2475680/xray_.log)

![20181013202421_1](https://user-images.githubusercontent.com/35204405/46908197-215a2f00-cf28-11e8-8118-fb46c3de74a1.jpg)

I have a save just before that happened. Since GitHub does not support .sav, I uploaded it to another file sharing site. Here it is: https://yadi.sk/d/OIvzJV_3hiBL7w

**Comment by SurDno** [comment]
Got same crash again on testing and trying to reproduce.
```
Expression    : fatal error
Function      : CScriptEngine::lua_error
File          : E:\priquel\sources\engine\xrServerEntities\script_engine.cpp
Line          : 180
Description   : <no expression>
Arguments     : LUA error: No such operator defined

lua stack:
  [ 1|-1] (no value):  
```
This time, I killed just one of them, the other one became an enemy for me, but got frozen and did not even try to kill me. The game crashed not immediately as the bloodsucker should have spawned, but after 20+ more seconds.
![image](https://user-images.githubusercontent.com/35204405/46908515-71d38b80-cf2c-11e8-8266-2a6982937038.png)

**Comment by SurDno** [comment]
While trying to reproduce the crash, got another strange behavior. I actually once got to a point in the cutscene when squad members start shooting to nowhere, just before bloodsucker spawns. But bloodsucker did not spawn and CS squad members continued shooting to nowhere.

**Comment by SurDno** [comment]
That's the save where I reproduced #1 bug, by the way, so there is no main quest in PDA and I am not in Clear Sky faction.

**Comment by Decane** [comment]
I am aware of grave issues with the sr_bloodsucker scheme. Incidentally, I made reference to it a few days ago in a ModDB comment:

>I won't release v1.1.3 officially until I've mustered up the willpower to fix the sr_bloodsucker scheme, which seems to have become glitchier with every new optimization I've introduced to the game.

The issues you are seeing are not limited to the marsh creature scene in the Swamp - they apply to all the other marsh creature appearances in the game, too, such as the one most players will encounter while escorting the Duty squad in Agroprom.

The scheme is poorly written and seems to be susceptible to race conditions, which have become more prevalent as I have made the game's scripts perform better. I do intend to fix it for the final release of v1.1.3.

**Comment by Decane** [comment]
Fixed in https://github.com/Decane/SRP/commit/8ea83cf84fb10059b7e0b774b996a204fc0331e4.

The problem was a race condition between `bloodsucker_attack:update(delta)` in sr_bloodsucker.script and `generic_object_binder:update(delta)` in bind_monster.script, where:

1. `bloodsucker_attack:update(delta)` would add the bloodsucker to bind_stalker.delayed_alife_release_objects.
2. (The problem:) `generic_object_binder:update(delta)` would delete the server entity even of **non-loaded** creatures (i.e. creatures created in the presiding game session) in bind_stalker.delayed_alife_release_objects the first time it runs.
3. `bloodsucker_attack:update(delta)` would attempt to call methods on the now-orphaned client object - one without a counterpart server entity.
